lib/multi_kafka_producer.rb

```diff
@@ -27,6 +27,7 @@ def self.publish(topic, *msgs_and_keys)
   KAFKAS = { kafka: 'jruby-kafka', poseidon: 'poseidon' }
 
   def self.default_adapter
+    return :null if Rails.env.test?
```

Comment:
Yes, all tests just stub out `MultiKafkaProducer.publish` but if you don't you'll end up actually hitting Kafka (which we probably almost never need in the test suite). It's no longer needed for this branch though, so I've taken it out. I might add a Test adapter later which remembers stuff sent to it, which means tests can use that instead of setting mock expectations whenever they check stuff sent into Kafka.